### ISSUE TITLE
[kinetic] can't update distribution due to various issues

### ISSUE THREAD

**Issue by mcwimpy** [Vanilla-OS/iso-builder#268]
### Issue Description

after installation of 22.10 (several alpha versions of boot medias crashed/froze or otherwise did not make or create a bootable system) i did about 37 times this first setup thing, and it fails each and every time anew. i tried updating in the hope a bug fixed tool would then download and let me finish the first setup, but it turned out, that the system does not want to do an update without the first setup having finished. so i'm kinda locked in a bug loop.

### Steps to Reproduce

- install
- let first setup fail (countless times)
- error message: failed to start transactional shell: open /partFuture/.system/:no such file or directory
- close first setup
- popup says 'command completed: /usr/bin/vanilla-first-setup'
- try 'apx update' (error 'do not run as root' with sudo, without it does nothing then complain about sudo not configured right: 
vatux@hpprobook6560b:§ sudo apx update
[sudo] password for vatux: 
Do not run Apx as root!
vatux@hpprobook6560b:§ apx update
sudo: /etc/sudo.conf is group writable
sudo: /usr/bin/sudo must be owned by uid 0 and have the setuid bit set
vatux@hpprobook6560b:§ 
)
- make screenshots and log files is all i can do now

### On what version of Vanilla OS this happens?

22.10

### Additional Information

vatux@hpprobook6560b:~$ vanilla-first-setup
WARNING:FirstSetup::Builder:failed to create log file: /etc/vanilla/first-setup.log
WARNING:root:No log will be stored.
INFO:FirstSetup::Builder:Performing display-condition: lspci | grep -i '.* nvidia .*'
INFO:FirstSetup::Builder:Step nvidia skipped due to display-conditions
INFO:FirstSetup::Builder:Performing display-condition: grep 'hypervisor' /proc/cpuinfo
INFO:FirstSetup::Builder:Step vm skipped due to display-conditions
TypeError: Must be number, not NoneType

(vanilla-first-setup:7205): Adwaita-CRITICAL **: 20:50:42.657: adw_carousel_get_nth_page: assertion 'n < adw_carousel_get_n_pages (self)' failed
INFO:FirstSetup::Processor:processing the following commands: 
export DEBIAN_FRONTEND=noninteractive
apt update
apt install -f
apt upgrade -y
apt install -y flatpak gnome-software-plugin-flatpak
!nextBoot flatpak remote-add --user --if-not-exists flathub https://flathub.org/repo/flathub.flatpakrepo
apt install -y software-properties-common
add-apt-repository --yes ppa:appimagelauncher-team/stable
apt install -y fuse3 libfuse2 appimagelauncher
!nextBoot flatpak install -y org.gnome.Calendar
!nextBoot flatpak install -y org.gnome.Characters || snap install gnome-characters
!nextBoot flatpak install -y org.gnome.Cheese || snap install cheese
!nextBoot flatpak install -y org.gnome.clocks || snap install gnome-clocks
!nextBoot flatpak install -y org.gnome.Connections
!nextBoot flatpak install -y org.gnome.Contacts || snap install gnome-contacts
!nextBoot flatpak install -y org.gnome.baobab
!nextBoot flatpak install -y org.gnome.Evince || snap install evince
!nextBoot flatpak install -y org.gnome.font-viewer
!nextBoot flatpak install -y org.gnome.eog || snap install eog
!nextBoot flatpak install -y org.gnome.Logs || snap install gnome-logs
!nextBoot flatpak install -y org.gnome.Maps
!nextBoot flatpak install -y org.gnome.TextEditor
!nextBoot flatpak install -y org.gnome.Weather || snap install gnome-weather
!nextBoot flatpak install -y org.libreoffice.LibreOffice || snap install libreoffice
!nextBoot flatpak install -y com.usebottles.bottles
!nextBoot flatpak install -y com.heroicgameslauncher.hgl
!nextBoot flatpak install -y net.lutris.Lutris
!nextBoot flatpak install -y org.gnome.Boxes || snap install gnome-boxes
!nextBoot flatpak install -y org.gnome.DejaDup || snap install deja-dup
!nextBoot flatpak install -y com.github.tchx84.Flatseal
!nextBoot flatpak install -y fr.romainvigier.MetadataCleaner
!nextBoot flatpak install -y com.github.flxzt.rnote
!nextBoot flatpak install -y de.haeckerfelix.Shortwave || snap install shortwave
!nextBoot flatpak install -y org.gnome.SoundRecorder
apt install -y timeshift
apt install -y ubuntu-restricted-addons ubuntu-restricted-extras
apt install -y apport
systemctl enable apport.service || true
apt update
apt install -f

(vanilla-first-setup:7205): VTE-WARNING **: 20:50:45.161: (../src/vtepty.cc:810):void vte_pty_spawn_with_fds_async(VtePty*, const char*, const char* const*, const char* const*, const int*, int, const int*, int, GSpawnFlags, GSpawnChildSetupFunc, gpointer, GDestroyNotify, int, GCancellable*, GAsyncReadyCallback, gpointer): runtime check failed: ((spawn_flags & ignored_spawn_flags()) == 0)

(vanilla-first-setup:7205): Adwaita-CRITICAL **: 20:51:02.674: adw_carousel_get_nth_page: assertion 'n < adw_carousel_get_n_pages (self)' failed

(vanilla-first-setup:7205): Adwaita-CRITICAL **: 20:51:22.689: adw_carousel_get_nth_page: assertion 'n < adw_carousel_get_n_pages (self)' failed

(vanilla-first-setup:7205): Adwaita-CRITICAL **: 20:51:42.705: adw_carousel_get_nth_page: assertion 'n < adw_carousel_get_n_pages (self)' failed

(vanilla-first-setup:7205): Adwaita-CRITICAL **: 20:52:02.721: adw_carousel_get_nth_page: assertion 'n < adw_carousel_get_n_pages (self)' failed

(vanilla-first-setup:7205): Adwaita-CRITICAL **: 20:52:22.737: adw_carousel_get_nth_page: assertion 'n < adw_carousel_get_n_pages (self)' failed

(vanilla-first-setup:7205): VTE-WARNING **: 20:52:26.471: vte_terminal_get_text: Passing a GArray to retrieve attributes is deprecated. In a future version, passing non-NULL as attributes array will make the function return NULL.

(vanilla-first-setup:7205): Adwaita-CRITICAL **: 20:52:42.756: adw_carousel_get_nth_page: assertion 'n < adw_carousel_get_n_pages (self)' failed

(vanilla-first-setup:7205): Adwaita-CRITICAL **: 20:53:02.775: adw_carousel_get_nth_page: assertion 'n < adw_carousel_get_n_pages (self)' failed

(vanilla-first-setup:7205): Adwaita-CRITICAL **: 20:53:22.792: adw_carousel_get_nth_page: assertion 'n < adw_carousel_get_n_pages (self)' failed

(vanilla-first-setup:7205): Adwaita-CRITICAL **: 20:53:42.812: adw_carousel_get_nth_page: assertion 'n < adw_carousel_get_n_pages (self)' failed

(vanilla-first-setup:7205): Adwaita-CRITICAL **: 20:54:02.833: adw_carousel_get_nth_page: assertion 'n < adw_carousel_get_n_pages (self)' failed

(vanilla-first-setup:7205): Adwaita-CRITICAL **: 20:54:22.854: adw_carousel_get_nth_page: assertion 'n < adw_carousel_get_n_pages (self)' failed

(vanilla-first-setup:7205): Adwaita-CRITICAL **: 20:54:42.873: adw_carousel_get_nth_page: assertion 'n < adw_carousel_get_n_pages (self)' failed

(vanilla-first-setup:7205): Adwaita-CRITICAL **: 20:55:02.889: adw_carousel_get_nth_page: assertion 'n < adw_carousel_get_n_pages (self)' failed

(vanilla-first-setup:7205): Adwaita-CRITICAL **: 20:55:22.905: adw_carousel_get_nth_page: assertion 'n < adw_carousel_get_n_pages (self)' failed

(vanilla-first-setup:7205): Adwaita-CRITICAL **: 20:55:42.921: adw_carousel_get_nth_page: assertion 'n < adw_carousel_get_n_pages (self)' failed

(vanilla-first-setup:7205): Adwaita-CRITICAL **: 20:56:02.937: adw_carousel_get_nth_page: assertion 'n < adw_carousel_get_n_pages (self)' failed

(vanilla-first-setup:7205): Adwaita-CRITICAL **: 20:56:22.953: adw_carousel_get_nth_page: assertion 'n < adw_carousel_get_n_pages (self)' failed

(vanilla-first-setup:7205): Adwaita-CRITICAL **: 20:56:42.966: adw_carousel_get_nth_page: assertion 'n < adw_carousel_get_n_pages (self)' failed

(vanilla-first-setup:7205): Adwaita-CRITICAL **: 20:57:02.982: adw_carousel_get_nth_page: assertion 'n < adw_carousel_get_n_pages (self)' failed

(vanilla-first-setup:7205): Adwaita-CRITICAL **: 20:57:22.997: adw_carousel_get_nth_page: assertion 'n < adw_carousel_get_n_pages (self)' failed

(vanilla-first-setup:7205): Adwaita-CRITICAL **: 20:57:43.016: adw_carousel_get_nth_page: assertion 'n < adw_carousel_get_n_pages (self)' failed
vatux@hpprobook6560b:~$ 

![Screenshot from 2023-11-17 20-51-18](https://github.com/Vanilla-OS/live-iso/assets/11300847/314d8518-866c-4e7f-9af8-08046d6ba36a)

![vanillaos_error_03](https://github.com/Vanilla-OS/live-iso/assets/11300847/6cd58c91-8bfa-4eeb-92a8-21056354405a)
![vanillaos_error_02](https://github.com/Vanilla-OS/live-iso/assets/11300847/4ca13372-108b-4e46-b9d3-86c9444d7cd9)
![vanillaos_error_01](https://github.com/Vanilla-OS/live-iso/assets/11300847/ee8249c1-4dd1-4ac9-b221-05ea32804726)

**Comment by mcwimpy** [comment]
as discussed on Discord / Vanilla OS Server / support channel
[https://discord.com/channels/1023243680829681704/1023245426385760267/1178753868545400862]
here the requested outputs of the blkid and mount commands:
[blkid.output.txt](https://github.com/Vanilla-OS/live-iso/files/13484324/blkid.output.txt)
[mount.output.txt](https://github.com/Vanilla-OS/live-iso/files/13484325/mount.output.txt)

**Comment by mirkobrombin** [comment]
Which build is this? R9?

**Comment by mcwimpy** [comment]
```
> vatux@hpprobook6560b:~# cat /etc/os-release 
> PRETTY_NAME="VanillaOS 22.10 all"
> NAME="VanillaOS"
> VERSION_ID="22.10"
> VERSION="22.10 all"
> VERSION_CODENAME="kinetic"
> ID=ubuntu
> ID_LIKE=debian
> HOME_URL="https://github.com/vanilla-os"
> SUPPORT_URL="https://github.com/vanilla-os"
> BUG_REPORT_URL="https://github.com/vanilla-os"
> PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
> UBUNTU_CODENAME="kinetic"
> vatux@hpprobook6560b:~#
```

**Comment by mcwimpy** [comment]
> Which build is this? R9?

dunno how to find this information- i sent the lsb info instead above

**Comment by mcwimpy** [comment]
![new-vanilla-error-2024-01-19](https://github.com/Vanilla-OS/live-iso/assets/11300847/97f82a83-9dfa-41a6-8a88-fa12027f447c)
![new-vanilla-error-2024-01-19-2](https://github.com/Vanilla-OS/live-iso/assets/11300847/ca42c8bd-e591-4052-baed-6f4af05adfa7)
i tried various combination of selections during this initial setup thing, but so far no luck.
i guess you got a bugfix online to download, but it won't do OS updates until after the initial program ran through
so i am like stuck with a chicken/egg issue :-)
if you got a package to download and install - plz let me know.

**Comment by mirkobrombin** [comment]
The given information does not let me debug. I feel the bug was fixed already but your installation looks broken in some points, in the last screenshot ABRoot does not find the structure it expect.

Which version of ABRoot is? Issue an `abroot --version` please.

**Comment by mcwimpy** [comment]
hello. sorry for the long wait. i wouldn't be surprised if i unknowingly damaged something and broke it. i have it on my testing laptop and ther linux distros run in parallel, i just installed this one cuz the alpha of v2 would freez on all my computers. the newest development is that i boot my somewhat or "somehow broken" installation and the first run app tells me "all is done". i click the button DONE and there are somewhat more apps (for eg painting) ....

i always why the software app never showed any system updates at all...

i can make a screenshot of that command you requested. gotta use mobile cuz i'm on my main laptop now (sorry if quality is poor) ... in short, it's v1.4.3.
![new-vanilla-error-2024-01-19-3](https://github.com/Vanilla-OS/live-iso/assets/11300847/250e5f87-f0fb-4db2-9260-6a8acb0b5aea)
